bpf/lib/nodeport.h

```diff
@@ -1167,29 +1182,27 @@ static __always_inline bool snat_v4_needed(struct __ctx_buff *ctx, __be32 *addr,
 			 */
 			einfo = lookup_ip4_egress_endpoint(ip4->saddr, ip4->daddr);
 			if (einfo) {
-				*addr = einfo->egress_ip;
-				*from_endpoint = true;
-				return true;
+				/* If the packet is a reply it means that the
+				 * connection did not originate from a pod,
+				 * and so we should skip SNAT (similar to how we
+				 * skipped forwarding the packet over the tunnel
+				 * to the egress gateway in bpf_lxc)
+				 */
+				if (!ct_is_reply) {
```

Comment:
If `ep` is NULL, `ct_is_reply` will always be false, even if it is actually reply traffic, because we will never execute `ct_is_reply4` above. Or am I missing something?

Reply:
We want to skip SNAT only if traffic is both coming from a local endpoint/pod and is reply traffic, so if the first condition is false (`ep == NULL`) we don't really care if it's reply traffic or not

Reply:
Ah, I get it. If it's coming from a remote node (i.e., `ep == NULL`), it can't be reply traffic anyway. I think the comment should clarify this as it's probably not going to be obvious to anyone who's not deeply familiar with the egress gateway.